core/services/orm/tree_traversal/RelationNode.php

```diff
@@ -209,6 +223,28 @@ public function toArray()
         }
         return $tree;
     }
+
+    /**
+     * Gets the IDs of all the model objects to delete; indexed first by model object name.
+     * @since $VID:$
+     * @return array
+     */
+    public function getIds()
+    {
+        if (empty($this->model_obj_nodes)) {
+            return [];
+        }
+        $ids = [
+            $this->related_model->get_this_model_name() => array_combine(
+                array_keys($this->model_obj_nodes),
+                array_keys($this->model_obj_nodes)
+            )
```

Comment:
same thing here... are IDs the best thing to use for the value?

You could populate the array using IDs for keys and an empty array for the values like:

```php
$this->related_model->get_this_model_name() => array_fill_keys(
    array_keys($this->model_obj_nodes),
    []
)
```